test/test_connection.py

```diff
@@ -28,6 +28,21 @@ def test_match_hostname_match(self):
         asserted_hostname = "foo"
         _match_hostname(cert, asserted_hostname)
 
+    def test_match_hostname_ipaddress_none(self):
+        cert = {"subjectAltName": [("DNS", "foo")]}
+        asserted_hostname = "foo"
+        with mock.patch("urllib3.util.ssl_match_hostname.ipaddress", None):
+            assert _match_hostname(cert, asserted_hostname) is None
+
+    def test_match_hostname_invalid_ipaddress(self):
```

Comment:
This test case shouldn't be necessary? There shouldn't be any case where the `ipaddress` module is a string.

Reply:
Yes, but this test covers https://github.com/urllib3/urllib3/blob/048c0ba15cec8c7dd787670bcafd243a54590982/src/urllib3/util/ssl_match_hostname.py#L126

Reply:
That's probably easily solved with a docstring. Having to work that out later is going to be difficult.

Reply:
Yeah, I'm not convinced we should test that branch, I don't think it's triggerable during regular execution. Let's add a `# Defensive:` comment to the `else` instead?

Reply:
Ok, i've removed the test and added `Defensive` in else block